src/bootstrap/test.rs

```diff
@@ -834,22 +834,68 @@ impl Step for RustdocJSNotStd {
     }
 }
 
-fn check_if_browser_ui_test_is_installed_global(npm: &Path, global: bool) -> bool {
+fn get_browser_ui_test_version_from_string(input: &String) -> Option<String> {
+    for line in input.lines() {
+        if line.contains(&" browser-ui-test@") {
+            if let Some(long_part) = line.rsplit(" browser-ui-test@").next() {
+                if let Some(version) = long_part.split(" ").next() {
+                    return Some(version.chars().filter(|c| *c == '.' || c.is_numeric()).collect());
+                }
+            }
+        }
+    }
+    None
+}
+
+fn check_if_browser_ui_test_is_installed_global(
+    npm: &Path,
+    target_version: &String,
+    global: bool,
+) -> bool {
     let mut command = Command::new(&npm);
     command.arg("list").arg("--depth=0");
     if global {
         command.arg("--global");
     }
-    let lines = command
+    let output = command
         .output()
-        .map(|output| String::from_utf8_lossy(&output.stdout).into_owned())
+        .map(|output| {
+            if output.status.success() {
+                String::from_utf8_lossy(&output.stdout).into_owned()
+            } else {
+                String::new()
+            }
+        })
         .unwrap_or(String::new());
-    lines.contains(&" browser-ui-test@")
+    if let Some(current_version) = get_browser_ui_test_version_from_string(&output) {
+        current_version == *target_version
+    } else {
+        false
+    }
 }
 
-fn check_if_browser_ui_test_is_installed(npm: &Path) -> bool {
-    check_if_browser_ui_test_is_installed_global(npm, false)
-        || check_if_browser_ui_test_is_installed_global(npm, true)
+fn get_browser_ui_test_version() -> std::io::Result<String> {
```

Comment:
This is a bit perillous to get this information like this...

Reply:
We could generate the Dockerfile from a template and then store the version in a separate file?